### PR TITLE
Allow headers to be passed to the WebSocket connection for VMServiceFlutterDriver

### PR DESCRIPTION
**Pull request by aubreyand**
## Description

Adds a headers parameter to FlutterDriver.connect. Updated FlutterDriver.connect to pass those headers to VMServiceFlutterDriver. Pass headers to the underlying WebSocket connections.
This change allows FlutterDriver to connect to a flutter instance on the other side of a reverse proxy that uses headers for destination resolution.

## Related Issues
None

## Tests

I added the following tests:

'connects with headers' to flutter_driver_test.dart

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

I had to change the setup for the flutter driver tests, but that was the only "test" that broke. I can write a migration guide if this should count as an existing test failing.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by dnfield — APPROVED** [review]
This change LGTM, but needs a migration guide written.  That's at https://github.com/flutter/website, there are some examples in https://github.com/flutter/website/tree/master/src/docs/release/breaking-changes

/cc @goderbauer @jonahwilliams who might see something I'm missing.

**Comment by jonahwilliams** [comment]
Why would this need a migration guide?

**Comment by dnfield** [comment]
Oh.  From offline discussion I was under the impression this broke a test or some internal code.  But if that's not the case this doesn't need migration guide.

**Comment by jonahwilliams** [comment]
This change seems fine as is. There are no tests broken by the addition, and nothing in the customer testing shard has gone red. 

FYI @angjieli in case there is a google3 implementation, but I think they all live in this repo.

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 1]
```diff
@@ -127,6 +127,10 @@ abstract class FlutterDriver {
   /// `isolateNumber` is set, as this is already enough information to connect
   /// to an isolate.
   ///
+  /// `headers` optionally specifies HTTP headers to be included in the
+  /// WebSocket connection. This is only used for VMServiceFlutterDriver
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:131]:
```suggestion
  /// [WebSocket] connection. This is only used for [VMServiceFlutterDriver]
```

to get cross links in our dart docs.

**Review by goderbauer — COMMENTED** [review]
This looks good to me as long as it doesn't break anything internally.

**Comment by goderbauer** [comment]
Talked to @angjieli offline. He things this should be fine.

**Pull request merged**